base/applications/cmdutils/wscript/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ add_typelib(ihost.idl)
 set_source_files_properties(rsrc.rc PROPERTIES OBJECT_DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/ihost.tlb)
 target_link_libraries(wscript uuid wine)
 set_module_type(wscript win32gui UNICODE)
-add_importlibs(wscript shell32 oleaut32 ole32 user32 advapi32 msvcrt kernel32 ntdll)
+add_importlibs(wscript shell32 oleaut32 ole32 user32 advapi32 shlwapi comctl32 msvcrt kernel32 ntdll)
```

Comment:
I don't see the problem of importing directly. This is a scripting host, no 3rd-party native code is going to throw a fit.

Windows will probably delay load the url handling. WSH was an optional component for Win95 and I just don't see the point of trying to replicate it for a reason that is no longer relevant.

As for comctl32, that was just to get the DSA functions. Replicating it on my own just to avoid some import is not worth it IMHO.

Reply:
I disagree, we should try to match the import of our target whereever possible. Including less than MS is fine, including more is most likely not a good idea anywhere.

Reply:
What good does it do if I just copy the code I need from comctl32? That just adds extra bloat and becomes a maintenance burden for zero benefit other than somehow being more NT5 compatible in ways that are not obserable.

As soon as the script uses the Shell.Application object or whatever, it's going to load shell32 and in turn shlwapi anyway.

Reply:
Do it like MS does it please. That is the purpose of ros, and is a good thing to act by default unless there are really profound reasons to act differently. I cannot see such reasons here. And depending on whatever additional stuff is rarely a good thing to do.

Reply:
If you want my approval for something, then it may make sense to split your tests from the implementations in a different PR. I don't have any complaints within the added tests-part.

Reply:
The wine version already links to shell32 (unlike Windows) and that pulls in shlwapi.